src/main/java/io/nats/client/ConnectionListener.java

```diff
@@ -70,12 +70,26 @@ public String toString() {
     }
 
     /**
+     * @deprecated use new api that gives additional details
      * Connection related events that occur asynchronously in the client code are
      * sent to a ConnectionListener via a single method. The ConnectionListener can
      * use the event type to decide what to do about the problem.
-     * 
+     *
      * @param conn the connection associated with the error
      * @param type the type of event that has occurred
      */
+    @Deprecated
     void connectionEvent(Connection conn, Events type);
+
+    /**
+     * Connection related events that occur asynchronously in the client code are
+     * sent to a ConnectionListener via a single method. The ConnectionListener can
+     * use the event type to decide what to do about the problem.
+     * @param conn the connection associated with the error
+     * @param type the type of event that has occurred
+     * @param uriDetails extra details about the uri related to this connection event
+     */
+    default void connectionEvent(Connection conn, Events type, String uriDetails) {
+        connectionEvent(conn, type);
+    }

```

Comment:
default impl required for backward compatibility. I've verified that there is no issue, I have lambda implementations of connection listener in unit tests, examples, tons of my scratch classes that are not checked in, etc. so I know this is safe to do and won't break anyone.